src/x509.rs

```diff
@@ -147,19 +165,95 @@ fn check_validity_expiration() {
 pub struct UniqueIdentifier<'a>(pub BitStringObject<'a>);
 
 impl<'a> TbsCertificate<'a> {
+    /// Get a reference to the map of extensions.
+    pub fn extensions(&self) -> &HashMap<Oid, X509Extension> {
+        &self.extensions
+    }
+
+    pub fn basic_constraints(&self) -> Option<(bool, &x509_extensions::BasicConstraints)> {
+        self.extensions.get(&oid!(2.5.29.19)).map(|ext| {
+            match ext.extension_type().unwrap() {
+                crate::x509_extensions::ExtensionType::BasicConstraints(ref bc) => (ext.critical, bc),
```

Comment:
Please add a `use crate::x509_extensions::ExtensionType` and simplify all type paths